examples/interpreter/physl.cpp

```diff
@@ -173,43 +234,78 @@ int main(int argc, char* argv[])
     }
 
     // Read the file containing the source code
-    auto positional_args = vm["positional"].as<std::vector<std::string>>();
+    std::vector<std::string> positional_args;
+    if (vm.count("positional") != 0)
+    {
+        positional_args = vm["positional"].as<std::vector<std::string>>();
+    }
 
     std::string code_source_name;
     std::string user_code;
     std::size_t first_index = 0;
-    if (vm.count("code") != 0)
+    std::vector<phylanx::ast::expression> ast;
+
+    if (vm.count("load-ast") != 0)
     {
-        // Execute code as given directly on the command line
-        user_code = vm["code"].as<std::string>();
-        code_source_name = "<command_line>";
+        auto path = vm["load-ast"].as<std::string>();
+        ast = load_ast(path);
+        code_source_name = "<ast_dump>";
     }
     else
     {
-        // Interpret first argument as the file name for the PhySL code
-        user_code = read_user_code(positional_args[0]);
-        code_source_name = fs::path(positional_args[0]).filename().string();
-        first_index = 1;
+        if (vm.count("code") != 0)
+        {
+            // Execute code as given directly on the command line
+            user_code = vm["code"].as<std::string>();
+            code_source_name = "<command_line>";
+        }
+        else if (positional_args.size() > 0)
+        {
+            // Interpret first argument as the file name for the PhySL code
+            user_code = read_user_code(positional_args[0]);
+            code_source_name = fs::path(positional_args[0]).filename().string();
+            first_index = 1;
+        }
+        else
+        {
+            std::cout << "No code was provided.\n";
+
+            return -1;
+        }
+
+        try
+        {
+            // Compile the given code into AST
+            ast = phylanx::ast::generate_ast(user_code);
+
+            // Apply transformation rules to AST, if requested
+            if (vm.count("transform") != 0)
+            {
+                std::string const transform_rules =
+                    read_user_code(vm["transform"].as<std::string>());
+
+                ast = phylanx::ast::transform_ast(
+                    ast, phylanx::ast::generate_transform_rules(transform_rules));
+            }
+
+            // Dump the AST to a file, if requested
+            if (vm.count("dump-ast") != 0)
+            {
+                dump_ast(ast, vm["dump-ast"].as<std::string>());
```

Comment:
If `--dump-ast` is used without any additional parameters (no file name) we could use the base file name of the input file, just replacing the extension to `.physl-ast` or just `.ast` (or similar).